carcara/src/checker/rules/pb_blasting.rs

```diff
@@ -169,8 +169,22 @@ pub fn pbblast_bvuge(RuleArgs { .. }: RuleArgs) -> RuleResult {
 ///
 /// The expected shape is:
 ///    `(= (bvule x y) (>= (- (+ sum_y) (+ sum_x)) 0))`
-pub fn pbblast_bvule(RuleArgs { .. }: RuleArgs) -> RuleResult {
-    Err(CheckerError::Unspecified)
+pub fn pbblast_bvule(RuleArgs { pool, conclusion, .. }: RuleArgs) -> RuleResult {
+    let ((x, y), ((sum_y, sum_x), constant)) =
+        match_term_err!((= (bvule x y) (>= (- sum_y sum_x) constant)) = &conclusion[0])?;
+
+    // Get the summation lists
+    let sum_x = get_pbsum(sum_x);
+    let sum_y = get_pbsum(sum_y);
+
+    // Check that the constant is 0
+    let constant: Integer = constant.as_integer_err()?;
+    rassert!(
+        constant == 0,
+        CheckerError::Explanation(format!("Non-zero constant {}", constant))
+    );
```

Comment:
You can use `match_term_err!` to check if the constant is 0:
```suggestion
    let ((x, y), ((sum_y, sum_x), ())) =
        match_term_err!((= (bvule x y) (>= (- sum_y sum_x) 0)) = &conclusion[0])?;

    // Get the summation lists
    let sum_x = get_pbsum(sum_x);
    let sum_y = get_pbsum(sum_y);
```